syntax = "proto3";
option go_package = "calc/proto";

service Calculator {
    rpc Add (CalcReq) returns (AddRes) {}
    rpc Sub (CalcReq) returns (SubRes) {}
    rpc Div (CalcReq) returns (DivRes) {}
    rpc Mul (CalcReq) returns (MulRes) {}
}

// for request variables
message CalcReq {
    float n1 = 1;
    float n2 = 2;
}

// for result/response variables
message AddRes {
    float r = 1;
}

message SubRes {
    float r = 1;
}

message DivRes {
    float r = 1;
}

message MulRes {
    float r = 1;
}